components/Elevation/tests/unit/MaterialElevationColorTests.m

```diff
@@ -296,4 +296,50 @@ - (void)testSmoothJumpBetweenElevationToAlphaForValuesCloseToOne {
                                           secondColor:resolvedSecondRGBColor];
 }
 
+- (void)testResolvingColorWithSameCurrentTraitCollectionAndPreviousTraitCollection {
+  if (@available(iOS 13.0, *)) {
+    // Given
+    CGFloat elevation = (CGFloat)10;
+    UIColor *darkColor = UIColor.blackColor;
+    UIColor *lightColor = UIColor.whiteColor;
+    UIColor *dynamicColor = [UIColor colorWithUserInterfaceStyleDarkColor:darkColor
+                                                             defaultColor:lightColor];
+
+    // When
+    UITraitCollection *previousTraitCollection =
+        [UITraitCollection traitCollectionWithUserInterfaceStyle:UIUserInterfaceStyleDark];
+    UITraitCollection *currentTraitCollection =
+        [UITraitCollection traitCollectionWithUserInterfaceStyle:UIUserInterfaceStyleDark];
+    UIColor *resolvedColor =
+        [dynamicColor mdc_resolvedColorWithTraitCollection:currentTraitCollection
+                                   previousTraitCollection:previousTraitCollection
+                                                 elevation:elevation];
+    // Then
+    XCTAssertEqualObjects(resolvedColor, dynamicColor);
+  }
+}
+
+- (void)testResolvingColorWithDifferenceCurrentTraitCollectionAndPreviousTraitCollection {
+  if (@available(iOS 13.0, *)) {
+    // Given
+    CGFloat elevation = (CGFloat)10;
+    UIColor *darkColor = UIColor.blackColor;
+    UIColor *lightColor = UIColor.whiteColor;
+    UIColor *dynamicColor = [UIColor colorWithUserInterfaceStyleDarkColor:darkColor
+                                                             defaultColor:lightColor];
+
+    // When
+    UITraitCollection *previousTraitCollection =
+        [UITraitCollection traitCollectionWithUserInterfaceStyle:UIUserInterfaceStyleDark];
+    UITraitCollection *currentTraitCollection =
+        [UITraitCollection traitCollectionWithUserInterfaceStyle:UIUserInterfaceStyleLight];
+    UIColor *resolvedColor =
+        [dynamicColor mdc_resolvedColorWithTraitCollection:currentTraitCollection
+                                   previousTraitCollection:previousTraitCollection
+                                                 elevation:elevation];
+    // Then
+    XCTAssertNotEqual(resolvedColor, dynamicColor);
```

Comment:
Can we test that the trait collection = what UIKit would return?

Reply:
Addressed in 75d9d11dfb3fd470687a570978baba739a411c59.